### PR TITLE
Extract tableau workbook tags as SystemTags

### PR DESCRIPTION
**Pull request by usefulalgorithm**
### 🤔 Why?

Tableau workbook can be tagged, our connector needs to support that.

### 🤓 What?

- Implemented parsing logic for the tags on workbooks.
- Modified unit test.

### 🧪 Tested?

- Tested locally.
- Unit test.

### ☑️ Checks

- [x] My PR contains actual code changes, and I have updated the version number in `pyproject.toml`.

### CODE REVIEW

**Comment by mars-lan** [comment]
To show clean diff, you can base your PR off `tsung-julii/sc-22604/add-config-option-to-exclude-tableau-personal` instead.

File: metaphor/tableau/extractor.py [hunk 1]
```diff
@@ -282,6 +288,18 @@ def _parse_dashboard(self, workbook: tableau.WorkbookItem) -> None:
 
         self._dashboards[workbook_id] = dashboard
 
+        if workbook.tags:
+            hierarchy = Hierarchy(
+                logical_id=HierarchyLogicalID(
+                    path=[DashboardPlatform.TABLEAU.value, workbook_id],
+                ),
+                system_tags=SystemTags(
+                    tags=[SystemTag(value=tag) for tag in sorted(workbook.tags)]
```
[the remark this answers is not in the document]
Comment (usefulalgorithm) [reply metaphor/tableau/extractor.py:297]: `Hierarchy` already contains the source in `logical_id.path`, I think it should be fine?

File: metaphor/tableau/extractor.py [hunk 2]
```diff
@@ -282,6 +288,18 @@ def _parse_dashboard(self, workbook: tableau.WorkbookItem) -> None:
 
         self._dashboards[workbook_id] = dashboard
 
+        if workbook.tags:
+            hierarchy = Hierarchy(
```
[the remark this answers is not in the document]
Comment (usefulalgorithm) [reply metaphor/tableau/extractor.py:292]: - `systemTags` is an entity of only `Hierarchy`, not for `Dashboard` and `VirtualView`.
- `workbook` here actually corresponds to a dashboard.
- It is indeed also possible to tag datasources. Do we want to support that?

File: metaphor/tableau/extractor.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mars-lan) [reply metaphor/tableau/extractor.py:297]: `systemTagSource `is a required field and currently only supports `SNOWFLAKE` .

File: metaphor/tableau/extractor.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mars-lan) [reply metaphor/tableau/extractor.py:292]: We need to add `systemTags` to all assets (`Dataset`, `Dashboard`, `VirtualView`).

File: metaphor/tableau/extractor.py [hunk 3]
```diff
@@ -332,13 +343,22 @@ def _parse_custom_sql_table(
         except Exception as e:
             logger.error(f"Unable to parse custom query for {custom_sql_table.id}: {e}")
 
+        system_tags = None
+        if custom_sql_table.column_tags:
```
Comment (mars-lan) [inline metaphor/tableau/extractor.py:347]: SystemTags are only for asset-level?

Comment (usefulalgorithm) [reply metaphor/tableau/extractor.py:347]: I thought we need to tag the upstream datasets as well?

Comment (mars-lan) [reply metaphor/tableau/extractor.py:347]: Are `column_tags` tags for columns?

Comment (mars-lan) [reply metaphor/tableau/extractor.py:347]: Yeah, but the point is what you're setting is table-level tags. I don't think we have models for column-level system tags yet.

Comment (usefulalgorithm) [reply metaphor/tableau/extractor.py:347]: I see, I'll remove that then. I thought column level tags are considered table tags as well

**Pull request merged**